### ISSUE TITLE
Registration leak when session closing throws exception

### ISSUE THREAD

**Issue by DZabavchik** [crossbario/crossbar#2130]
Memory leaks fixed by #2098 unmasked smaller memory leaks in [dealer.py](https://github.com/crossbario/crossbar/blob/master/crossbar/router/dealer.py). 

Dealer.detach() iterates through invocations in flight where session is a caller or callee, performing inline `router.send` or `transport.send`.  If that throws exception the loop is interrupted, invocations are not removed resulting in memory leak as well as caller(s) not receiving `CANCELED`
